src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.4.0.0.cs

```diff
@@ -150,6 +151,492 @@ public static void ClientBaseOfT_Sync_RoundTrip_Check_CommunicationState()
 
     [WcfFact]
     [OuterLoop]
+    public static void ClientBaseOfT_Sync_Open_Close_Factory_And_Proxy_CommunicationState()
+    {
+        MyClientBase client = null;
+        IWcfServiceGenerated serviceProxy = null;
+        ChannelFactory<IWcfServiceGenerated> factory = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            CustomBinding customBinding = new CustomBinding();
+            customBinding.Elements.Add(new TextMessageEncodingBindingElement());
+            customBinding.Elements.Add(new HttpTransportBindingElement());
+
+            string endpoint = Endpoints.HttpSoap12_Address;
+            client = new MyClientBase(customBinding, new EndpointAddress(endpoint));
+            factory = client.ChannelFactory;
+
+            // *** VALIDATE *** \\
+            Assert.True(CommunicationState.Created == client.State,
+                        String.Format("Expected client state to be Created but actual was '{0}'", client.State));
+
+            Assert.True(CommunicationState.Created == factory.State,
+                        String.Format("Expected channel factory state to be Created but actual was '{0}'", factory.State));
+
+            // Note: both the full framework and this NET Core version open the channel factory
+            // when asking for the internal channel.  Attempting to Open the ClientBase
+            // after obtaining the internal channel with throw InvalidOperationException attempting
+            // to reopen the channel factory.  Customers don't encounter this situation in normal use
+            // because access to the internal channel is protected and cannot be acquired.  So we
+            // defer asking for the internal channel in this test to allow the Open() to follow the
+            // same code path as customer code.
```

Comment:
Thanks for this comment, this is the sort of thing that I think makes our test bed valuable as a "how to" reference.